src/applications/personalization/profile/components/Profile.jsx

```diff
@@ -304,17 +304,12 @@ const mapStateToProps = state => {
 
   const hasLoadedEDUPaymentInformation = eduDirectDepositInformation(state);
 
-  const hasLoadedTotalDisabilityRating =
-    state.totalRating?.totalDisabilityRating || state.totalRating?.error;
-
   const hasLoadedAllData =
     hasLoadedFullName &&
     hasLoadedMHVInformation &&
     hasLoadedPersonalInformation &&
     hasLoadedMilitaryInformation &&
-    (shouldFetchTotalDisabilityRating
-      ? hasLoadedTotalDisabilityRating
-      : true) &&
+    (shouldFetchTotalDisabilityRating ? !state.totalRating?.loading : true) &&
```

Comment:
This seems technically not _quite_ right. there could be a moment when `state.totalRating` is not defined in which case it'll be treated as though the data had loaded, right? Ternaries that use the leading `!` always make my head spin! BUT, even in that super edge case that'll last virtually no time, it's not going to result in `hasLoadedAllData` to be `true`

Reply:
Awesome catch, `!undefined` is truthy.. updated! 💯